modules/openapi-generator/src/main/resources/aspnetcore/3.0/ModelsProject.csproj.mustache

```diff
@@ -13,25 +13,32 @@
     <OutputType>Library</OutputType>
     <AssemblyName>{{modelPackage}}</AssemblyName>
     <PackageId>{{modelPackage}}</PackageId>
+{{#useCentralizedPackageVersionManagement}}
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+{{/useCentralizedPackageVersionManagement}}
+{{#centralizedPackageVersionManagementOptOut}}
+    <ManagePackageVersionsCentrally>false</ManagePackageVersionsCentrally>
+{{/centralizedPackageVersionManagementOptOut}}
+
```

Comment:
If I understand you correctly that would mean when setting it to false it has a different behaviour than dont setting it at all?
Personally I dont like this. As a user I would expect that if I dont set it at all it would do the same as false. If you want to have only one option though, we could use a enum with "true", "opt-out", "default"

Reply:
> As a user I would expect that if I dont set it at all it would do the same as false

we can default the option to false

Reply:
We could, but not having the property in the .csproj file and setting the property to false are very different things

Reply:
OK

> we could use a enum with "true", "opt-out", "default". Not setting it would use the default option

what about having 3 values in a single option instead as you suggested earlier?

Reply:
I am currently trying to implement it but having difficulties working with an enum

Reply:
take a look at the library option, which also uses enum

Reply:
Are the enum values validated automatically or do I have to do it myself?

Reply:
I pushed a commit with an initial implementation. the issue I am currently having is that I can put in any value I want and the generator will just put the string in the project without making sure that the value is one of the enum values

Reply:
Should be automatic

Reply:
I tried everything I could think of but I still cant get it to validate that the value is one of the enums. How do we procede with this pr?

Reply:
Thanks :)